src/requirements/data/majors/dea.ts

```diff
@@ -0,0 +1,221 @@
+    source:
+      'http://courses.cornell.edu/preview_program.php?catoid=41&poid=19856#deamajorrequirements',
+    checker: (course: Course): boolean =>
+      ['CA', 'HA', 'LA/LAD'].some(
+        distribution => course.catalogDistr?.includes(distribution) ?? false
+      ),
+    operator: 'or',
+    fulfilledBy: 'courses',
+    minCount: 1,
+  },
+  {
+    name: 'Statistics',
+    description:
+      'Must be taken at Cornell, AP Statistics is not accepted. Choose one of the following: ' +
+      'PAM 2100, AEM 2100, ILRST/STSCI 2100, or PSYCH 2500',
+    source:
+      'http://courses.cornell.edu/preview_program.php?catoid=41&poid=19856#deamajorrequirements',
+    checker: includesWithSubRequirements(['PAM 2100', 'AEM 2100', 'ILRST 2100', 'PSYCH 2500']),
+    operator: 'or',
+    fulfilledBy: 'courses',
+    minCount: 1,
+  },
+  {
+    name: 'Natural Science I',
+    description:
+      'Choose one of the following options. If AP isn’t used to satisfy the requirement, then the course must be taken at Cornell. No lab is required.',
+    source:
+      'http://courses.cornell.edu/preview_program.php?catoid=41&poid=19856#deamajorrequirements',
+    fulfilledBy: 'toggleable',
+    fulfillmentOptions: {
+      Biology: {
+        description: 'Take 1: BIOG 1140, BIOG 1440, BIOMG 1350, BIOEE 1610 or BIOG 1445',
+        checker: includesWithSubRequirements([
+          'BIOG 1140',
+          'BIOG 1440',
+          'BIOMG 1350',
+          'BIOEE 1610',
+          'BIOG 1445',
+        ]),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+      // TODO: Placeholder for AP Bio
+      'AP Biology score of 5': {
+        description: 'AP Biology score of 5',
+        checker: includesWithSubRequirements(['BIOG 1140']),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+      Chemistry: {
+        description: 'Take 1: CHEM 1560 or CHEM 2070 or CHEM 2080',
+        checker: includesWithSubRequirements(['CHEM 1560', 'CHEM 2070', 'CHEM 2080']),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+      // TODO: Placeholder for AP Chem
+      'AP Chemistry score of 5': {
+        description: 'AP Chemistry score of 5',
+        checker: includesWithSubRequirements(['CHEM 2080']),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+      Physics: {
+        description: 'Take 1: PHYS 1101, PHYS 2207, PHYS 1102, or PHYS 2208',
+        checker: includesWithSubRequirements(['PHYS 1101', 'PHYS 2207', 'PHYS 1102', 'PHYS 2208']),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+      // TODO: Placeholder for AP Physics
+      'AP Physics score of 5': {
+        description: 'AP Physics score of 5',
+        checker: includesWithSubRequirements(['PHYS 1101']),
+        counting: 'courses',
+        operator: 'and',
+        minCount: 1,
+      },
+    },
```

Comment:
should this be automatically handled by the above requirement?

Reply:
I'll double check with @benjamin-shen 
I wasn't sure how users would choose to accept/not accept their AP credit without toggle-able requirements.

Reply:
I'm not sure if I'm interpreting your question correctly.

Equivalent course credit for AP exams should in general be automatically accepted. I can't think of any scenarios where there are multiple fulfillment strategies for AP/IB equivalent courses. In this case, with toggleable requirements (if I'm understanding correctly), I believe users can choose Physics and their AP credit (or IB credit, or any transfer credit that has an acceptable course id) would automatically apply.

More in-depth explanation: Sam is right, the "Physics" fulfillment option should automatically cover "AP Physics score of 5". When running the requirements json generator, the Physics checker `includesWithSubRequirements(['PHYS 1101', ...])` should add the course id for PHYS 1101 to the `decorated-requirements.json`. Then, if the user takes AP Physics I, `getCourseEquivalentsFromUserExams` should add the equivalent course (with the PHYS 1101 course id) to the courses taken, which then behaves as expected in the requirements graph/algorithm. I'm not exactly sure how the toggleable requirements work but naturally it seems like it would be the same logic: the checker would convert the course code to the course id, which would then match AP Physics' (or whatever exam's) equivalent course's course id.

So @einc I think you can just get rid of all the // TODO: Placeholder's.

Reply:
That makes sense! Sorry one more question, so if the user chooses not to use AP credit for a course, what would that look right? I think ideally, we would want to give users the option (if possible)? 

For example, a user could have AP CS credit, but choose not to take it and instead take CS 1110.
Would pressing "reset" remove the AP credit (or not because the transfer credit is a part of the requirements precomputation)? Or would the user have to remove the AP credit during onboarding?

Reply:
I don't think any of that is actually (explicitly) implemented. How *should* that work (on a high level)?

Currently, to forfeit the entire ap exam, the user would have to remove it from the onboarding page (because if they're adding it, we should assume they want at least some credit from it?). The reset button doesn't do anything right now but maybe it can remove the exam from the user on Firebase?

If there are cases where we need more control (eg. choose not to use an exam's equivalent course for one requirement, but use it for another requirement), I imagine we would need to update the way exams are stored (maybe with an additional `requirementsFulfilled` field) and use that to change the way course equivalents are generated (maybe change the `CourseTaken` type). Then, AP/IB equivalent courses would be able to act more like regular courses. But I'm pretty sure the granularity for regular courses (the "backend" to Hahnbee's work on the new add course modal) isn't implemented yet. So this is definitely a next-semester solution, unless someone has any better ideas.